simplefs/simplefs.go

```diff
@@ -950,15 +1173,42 @@ func (k *SimpleFS) SimpleFSCancel(_ context.Context, opid keybase1.OpID) error {
 // SimpleFSCheck - Check progress of pending operation
 // Progress variable is still TBD.
 // Return errNoResult if no operation found.
-func (k *SimpleFS) SimpleFSCheck(_ context.Context, opid keybase1.OpID) (keybase1.Progress, error) {
+func (k *SimpleFS) SimpleFSCheck(_ context.Context, opid keybase1.OpID) (keybase1.OpProgress, error) {
 	k.lock.RLock()
 	defer k.lock.RUnlock()
```

Comment:
Won't this read lock cause the read/write loops to block on updating the progress? We could slow down transfers a bit. I'm wondering if a channel approach would be better.

Reply:
I can't imagine it would matter to perceived performance, given the speed of KBFS uploads/downloads and the fact that this will be called maybe a few times a second at most and doesn't block on anything network-related, but if it becomes a problem, sure we can make a goroutine-based channel thingy.  I think that's overkill for now though.